src/main/java/uk/gov/pay/ledger/agreement/dao/AgreementDao.java

```diff
@@ -87,6 +89,21 @@ public class AgreementDao {
             "FROM agreement a " +
             ":searchExtraFields";
 
+    private static final String SELECT_ASSOCIATED_EVENTS = "SELECT results.* FROM " +
+            "(" +
+                "SELECT e.id, e.sqs_message_id, e.service_id, e.live, 'payment_instrument' AS resource_type_name, " +
+                "e.resource_external_id, e.parent_resource_external_id, e.event_date, e.event_type, e.event_data " +
+                "FROM payment_instrument pi " +
+                "JOIN event e ON e.resource_external_id = pi.external_id " +
+                "WHERE pi.agreement_external_id = :externalId " +
+                "UNION SELECT " +
+                "e.id, e.sqs_message_id, e.service_id, e.live, 'agreement' AS resource_type_name, e.resource_external_id, " +
+                "e.parent_resource_external_id, e.event_date, e.event_type, e.event_data " +
+                "FROM event e " +
+                "WHERE e.resource_external_id = :externalId " +
+            ") AS results " +
+            "ORDER BY results.event_date";
```

Comment:
Can the order (asc, desc) be specified explicitly so it is clear when reading the query?

Reply:
Happy to make this more explicit - done.